test/e2e/layering_test.go

```diff
@@ -0,0 +1,98 @@
+	require.Nil(t, err)
+	defer cs.ImageStreams(constants.MCONamespace).Delete(context.TODO(), imageStreamName, metav1.DeleteOptions{})
+
+	// push a build to the image stream
+	buildConfig := &buildv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: buildName,
+		},
+		Spec: buildv1.BuildSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Type: "Git",
+					Git: &buildv1.GitBuildSource{
+						URI: "https://github.com/cgwalters/fcos-derivation-example",
+						Ref: "rhcos",
+					},
+				},
+				Strategy: buildv1.BuildStrategy{
+					DockerStrategy: &buildv1.DockerBuildStrategy{},
+				},
+				Output: buildv1.BuildOutput{
+					To: &v1.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: imageStreamName + ":latest",
+					},
+				},
+			},
+		},
+	}
+	_, err = cs.BuildV1Interface.Builds(constants.MCONamespace).Create(context.TODO(), buildConfig, metav1.CreateOptions{})
+	require.Nil(t, err)
+	defer cs.BuildV1Interface.Builds(constants.MCONamespace).Delete(context.TODO(), buildName, metav1.DeleteOptions{})
+	helpers.WaitForBuild(t, cs, buildConfig.ObjectMeta.Name)
+
+	// pick a random worker node
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/infra")
+	defer unlabelFunc()
+	infraNode := helpers.GetSingleNodeByRole(t, cs, "infra")
+
+	// Run commands on the node:
+	// oc login -u kubeadmin -p <password_from_install_log> --insecure-skip-tls-verify https://api-int.<cluster_name>.<base_domain>:6443
+	// podman login -u kubeadmin -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000
+	imageRegistry := "image-registry.openshift-image-registry.svc:5000"
+	password := os.Getenv("PASSWORD")
+	clusterURL := os.Getenv("URL")
+	ocLogin := helpers.ExecCmdOnNode(t, cs, infraNode, "chroot", "/rootfs", "oc", "login", "-u", "kubeadmin", "-p", password, "--insecure-skip-tls-verify", fmt.Sprintf("https://api-int.%s:6443", clusterURL))
+	require.Contains(t, ocLogin, "Login successful.")
+	whoami := strings.TrimSuffix(helpers.ExecCmdOnNode(t, cs, infraNode, "chroot", "/rootfs", "oc", "whoami", "-t"), "\n")
+	podmanLogin := helpers.ExecCmdOnNode(t, cs, infraNode, "chroot", "/rootfs", "podman", "login", "-u", "kubeadmin", "-p", whoami, imageRegistry)
+	require.Equal(t, "Login Succeeded!\n", podmanLogin)
+
+	// rpm-ostree rebase --experimental ostree-unverified-image:docker://image-registry.openshift-image-registry.svc.cluster.local:5000/openshift-machine-config-operator/test-boot-in-cluster-image-build
+	imageURL := fmt.Sprintf("ostree-unverified-image:docker://%s/%s/%s", imageRegistry, constants.MCONamespace, imageStreamName)
+	rebase := helpers.ExecCmdOnNode(t, cs, infraNode, "chroot", "/rootfs", "rpm-ostree", "rebase", "--experimental", imageURL)
+	changesQueued := "Changes queued for next boot. Run \"systemctl reboot\" to start a reboot"
+	require.Contains(t, rebase, changesQueued)
```

Comment:
I know this is just a test, but I would like the ability to change the English text output from rpm-ostree without breaking the MCO's test suite.

I think we can probably just drop this chunk.  But if we *do* want to be explicit, we can use `rpm-ostree status --json` and check for a queued pending deployment.  The MCD has code for this already.  See also https://github.com/coreos/rpm-ostree/issues/2389

Reply:
Kinda feels like overkill but added something to call `rpm-ostree status --json` both here and after rollback